src/doc/en/developer/portability_testing.rst

```diff
@@ -949,49 +949,68 @@ Automatic testing on multiple platforms on GitHub Actions
 =========================================================
 
 The Sage source tree includes a default configuration for GitHub
-Actions that runs our portability tests with tox on a multitude of
-platforms on every pull request and on every push of a tag (but not of
-a branch) to a repository for which GitHub Actions are enabled.
-
-In particular, it automatically runs on our main repository on every
-release tag.
-
-This is defined in the files `$SAGE_ROOT/.github/workflows/tox*.yml
-<https://github.com/sagemath/sage/tree/develop/.github/workflows/tox.yml>`_.
-
-An additional GitHub Actions workflow for testing on Cygwin, not based
-on tox, is defined in the files
-`$SAGE_ROOT/.github/workflows/ci-cygwin*.yml
-<https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-cygwin-standard.yml>`_.
+Actions that runs our portability tests on a multitude of platforms on
+every push of a tag (but not of a branch) to a repository for which
+GitHub Actions are enabled.
+
+In particular, it automatically runs on our main repository sagemath/sage
+on every release tag.
+
+This is defined in the files
+- `$SAGE_ROOT/.github/workflows/ci-linux.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-linux.yml>`_
+  (which calls `$SAGE_ROOT/.github/workflows/docker.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/docker.yml>`_),
+- `$SAGE_ROOT/.github/workflows/ci-macos.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-macos.yml>`_, and
+- `$SAGE_ROOT/.github/workflows/ci-cygwin-standard.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-cygwin-standard.yml>`_
+  (which calls `$SAGE_ROOT/.github/workflows/cygwin.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/cygwin.yml>`_).
 
 GitHub Actions runs these build jobs on 2-core machines with 7 GB of
 RAM memory and 14 GB of SSD disk space, cf.
 `here <https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources>`_,
-and has a time limit of 6h per job. This is just barely enough for a
-typical ``minimal`` build followed by ``make ptest`` to succeed; and
-plenty of time for a typical ``standard`` build to succeed.
+and has a time limit of 6h per job. This could be just barely enough for a
+typical ``minimal`` build followed by ``make ptest`` to succeed; for
+added robustness, we split it into two jobs. Our workflow stores
+Docker images corresponding to various build phases within these two
+jobs on `GitHub Packages <https://github.com/features/packages>`_ (ghcr.io).
+
+Build logs can be inspected during the run and become available as
+"artifacts" when all jobs of the workflow have finished.  Each job
+generates one tarball.  "Annotations" highlight certain top-level
+errors or warnings issued during the build.
+
+In addition to these automatic runs in our main repository, all Sage
```

Comment:
I don't think its a good idea to advice people to run the workflows in their own forks. This a old workaround from the time where there was no easy way to run github workflows via trac. Instead I would propose to add labels to run the workflows, similarly how the conda workflow works.

Reply:
No, that's exactly what we want -- so that such tests do not clog the project's Actions runners.

Reply:
Don't we have enough parallel workflows now that we have a github team account? If a person only has a free github account, then it takes quite a bit of time until the complete matrix is finished since only a couple of them run in parallel - it also blocks other workflows in other projects. (Also not everyone wants to have 3000 published "packages": https://github.com/mkoeppe?tab=packages)

Reply:
We have 60 runners, and the ci-linux workflow, running 24 jobs in parallel, takes a few days. I think that's too much for unleashing it on the project's runner pool.

Reply:
I think 24 jobs each taking a couple of hours (why would they take 'a few days'? its only compiling + running tests) should be okay with 60 runners. If not, we can remove some of the legacy systems and older python versions. I'm also a bit confused since here https://github.com/sagemath/sage/pull/35403#issuecomment-1493037240 you denied that 'we might be hitting limits of what we have available on GitHub Actions.'

Either way, we cannot expect people to activate actions in their fork. I also don't know of any other open source project that would do this.

Reply:
No, 24 is the maximum of parallel jobs. It runs many more jobs.

Reply:
> you denied that 'we might be hitting limits of what we have available on GitHub Actions.'

Context. We are not hitting limits when only every release tag triggers this heavy-weight workflow. But we would be in trouble if 3 of those would be run in parallel; it would stall all other types of workflows.

Reply:
> Either way, we cannot expect people to activate actions in their fork

And nobody has to do it unless they want to.

Reply:
> why would they take 'a few days'?

Just look at it: https://github.com/sagemath/sage/actions/workflows/ci-linux.yml

Reply:
I'm not convinced that its a good idea to put this in the docs. If someone wants to run it in their fork, she is welcome to do this - but I don't think we should document and support this. 

In total it might take a few days, but each workflow takes only a couple of hours and thus only blocks a runner for a few hours. Moreover, due to the staged setup not all parallel runers are consumed at the same time (e.g. currently there are only 5), so that other workflows could be run in between.

Reply:
It's already in the docs, and I put it there for an important purpose: Namely to empower developers to learn about portability work, and that includes running Actions by themselves, not just seeing them run by the magic central repository.